source/Aconcagua-Quantitative-Analysis/InternationalSystemOfUnits.class.st

```diff
@@ -0,0 +1,668 @@
+"
+The International System of Units, denoted SI in all languages, was adopted by the 11th International
+Conference on Weights and Measures, CGPM [1960] (Conférence générale des poids et mesures). The SI is
+a coherent system of units with respect to the International System of Quantities.
+
+The SI comprises
+- base units, and
+- derived units
+that together form the coherent system of SI units.
+"
+Class {
+	#name : 'InternationalSystemOfUnits',
+	#superclass : 'Object',
+	#classVars : [
+		'Units'
+	],
+	#category : 'Aconcagua-Quantitative-Analysis',
+	#package : 'Aconcagua-Quantitative-Analysis'
+}
+
+{ #category : 'class initialization' }
+InternationalSystemOfUnits class >> addBaseUnitBasedOn: dimension singularNamed: singularName pluralNamed: pluralName symbol: symbol as: key [
+
+  <ignoreForCoverage>
+  | unit |
+  unit := BaseUnitOfMeasure
+            basedOn: dimension
+            singularNamed: singularName
+            pluralNamed: pluralName
+            symbol: symbol.
+  Units bind: key asSymbol to: unit.
+  ^ unit
+]
+
+{ #category : 'class initialization' }
+InternationalSystemOfUnits class >> addMultipleFor: baseUnit prefixedBy: namePrefix and: symbolPrefix raisedTo: exponent as: baseUnitKey [
+
+  <ignoreForCoverage>
+  | unit |
+  unit := MultipleUnitOfMeasure
+            basedOn: baseUnit
+            prefixedBy: namePrefix
+            symbolPrefixedBy: symbolPrefix
+            raisedTo: exponent.
+  Units bind: ( namePrefix , baseUnitKey ) asSymbol to: unit
```

Comment:
Why is `,` used for contatenating instead of `expandMacrosWith:` ?

Reply:
Because it's simpler and it is called only when defining the namespaces, so it doesn't affect runtime performance.